install-build-deps.sh

```diff
@@ -33,4 +33,4 @@ go install \
 # a lot of work, so we install all of those from the release tarball
 METALINTER_VERSION="3.0.0"
 curl --silent --location "https://github.com/alecthomas/gometalinter/releases/download/v${METALINTER_VERSION}/gometalinter-${METALINTER_VERSION}-${OSARCH}.tar.gz" \
-  | tar -x -z -C "${GOBIN}" --strip-components 1
+  | tar -x -z -C "${GOBIN%%:*}" --strip-components 1
```

Comment:
if `GOPATH` equals `/bin/go:$HOME/go` then [`GOBIN=/bin/go:$HOME/go/bin`](https://github.com/weaveworks/eksctl/pull/1632/files#diff-bad7ead3a3d2710cd37d93de0fa7ad24R7) and your parameter expansion `"${GOBIN%%:*}"` leads to `/bin/go` but it should be `/bin/go/bin` in my opinion.

Shouldn't the GOBIN assignment further up be changed to
```
GOBIN="${GOPATH%%:*}/bin"
```

Reply:
Sorry my example GOPATH was flipped. _Usually_ I see people add to their GOPATH with preferred (personal) directories at the beginning of the path to take precedence. This change would assume that they want to use the first directory in their GOPATH.

We can't change GOBIN earlier in the file because it's using `go env GOPATH` instead of `$GOPATH`. I might be the worst person to ask why we would use one vs the other here.

I'm fine making the change at that earlier declaration but we would still need to use the substitution in the tar command in case someone already had GOBIN set with multiple paths (see the if statement on line 6).

Reply:
> This change would assume that they want to use the first directory in their GOPATH.

My comment still stands, though. The code initially meant to append `/bin` to the `GOPATH` but your change does not do that anymore (in the case where `GOPATH` has multiple directories). It just picks the first in `GOPATH` without appending `/bin`.

> We can't change GOBIN earlier in the file because it's using go env GOPATH instead of $GOPATH. I might be the worst person to ask why we would use one vs the other here.

From what I can tell it's the same thing. You can just use `$GOPATH` instead if you want to keep bash's parameter expansion (or then assign `go env GOPATH` to a variable first)

> I'm fine making the change at that earlier declaration

Please do, thanks!

Reply:
Sorry about that. I see what you mean now. I think the latest push will work. It uses the *first* path in GOPATH and appends /bin